### ISSUE TITLE
Missing progress indication at initial start with huge collection

### ISSUE THREAD

**Issue by madsrh** [ubuntu-flutter-community/musicpod#281]
Very very low priority I know.
This could fall under the same title as "Slow initial start with huge collection". As eboye mentioned, the initial start takes a while with a huge local music collection. The spinner is good, but it just sits there. Is it possible to write the output? like "Indexing 324 out of 41.233". Or would be an idea to add some timed messages like:

"Hang on while we're indexing your local audio"
"Still working on it..."
"Wow! You've got a lot of music"
"There's some great songs in your music collection. Any second now."

🤷‍♂️ You get my point 😄

**Comment by Feichtmeier** [comment]
Good idea with the nice message
Outputting every file would probably slow down the process then more
I'll start with a better message :)

**Comment by Feichtmeier** [comment]
Tried some things here 

The import and metadata extraction is made in a second thread to keep the UI render at normal FPS meanwhile 

When I forward the information from one thread to the other I slow down the import :)

So I think we should keep it like this. Wdyt?

In the meantime I've switched to a much faster library to read metadata and also improved the code. It's always hard to check this with my 3k MP3s but it's very fast now, about 0.5-2 seconds depending if I'm on the Mac or the old Ubuntu book

**Comment by madsrh** [comment]
> So I think we should keep it like this. Wdyt?

You are absolutely right. This is nitpicking at it's best and a onetime task for the user. Everything looks sooooo slick 🥇 

I can't speak for the huge music collection (_mine is on another computer_), but I tried adding a USB stick with a few albums and it 
 does takes a few seconds with no confirmation that the folder was added successful. But I'm fine with us closing the issue as mentioned above.



https://github.com/ubuntu-flutter-community/musicpod/assets/3986894/2c59e2ef-9f9c-46de-b26b-a52fb70ee19c
